### ISSUE TITLE
Support state machine instance list in visual debugger

### ISSUE THREAD

**Issue by kmozsi** [ELTE-Soft/txtUML#534]
Supporting multiple instances would be desired, but currently we can debug only one instance of each state machine.

**Comment by djnemeth** [comment]
Due to limitations of Papyrus, the team have decided to support this feature only on JavaScript diagrams. As a result, this now depends on #576.

**Comment by djnemeth** [comment]
Postponed.

**Comment by szokolai-mate** [comment]
Updated description of this issue:

Currently, the JavaScript debug view only supports displaying one state machine diagram at a time.
The DiagnosticsServer already supplies the browser client with runtime information, which consists of all active state machines, thier current active state and their unique ID. We can also extend this with the runtime name of the state machine, given by the user.

Decisions to make:

- How do we display these state machine instances, which are theoratically infinite in number?
    - On different tabs (like seperate state machine models)?
        - This way, we would still only see one state machine at a time.
    - On one tab?
        - How do we merge the SVG canvases?
        - How do we arrange these canvases?
        - Reduce the drawn elements by applying a filter to which state machine instances are displayed?

**Comment by szokolai-mate** [comment]
After a team discussion, we decided on the following:

The "display unit" is one class: we can simultaneously see all instances of **one class**, if we choose to, and we can change between these views by selecting a different state machine diagram.

Features to achieve:

Ability to select which instances we wish to display
Displaying multiple instances
    either by merging the input, or merging the SVG elements generated by JointJS
Parameterizing the generation of the imput for said instances: name and id
Animating the instances: filtering animated state machines by id
